### PR TITLE
Ability to refresh mqtt password at every reconnection

### PR DESCRIPTION
**Pull request by leandronsp**
Hi, we have a scenario where every time we generate a password to the MQTT, this password may have some security constraints, i.e expiration time, depending on the MQTT module implementation.

Sometimes the `tcp` connection may be closed or the `emqttc` process might fail for any reason, then we'd like to use the same process to retry over again but using a different password. 

This patch adds the ability to send a `password_refresher` to the mqtt connection opts, a simple function which will be called at every connect/reconnect.

### Usage
```
PasswordRefresher = fun() -> <<"logic to generate a new password">> end,
{ok, C} = emqttc:start_link([{host, "localhost"}, {password_refresher, PasswordRefresher}]).
```

### CODE REVIEW

**Comment by leandronsp** [comment]
cc @emqplus

File: README.md [hunk 1]
```diff
@@ -111,6 +111,7 @@ Connect to MQTT Broker:
                    | {proto_ver, mqtt_vsn()}
                    | {username, binary()}
                    | {password, binary()}
+                   | {password_refresher, fun()}
```
Comment (picaoao) [inline README.md]: fun() | undefined

File: src/emqttc_protocol.erl [hunk 2]
```diff
@@ -119,6 +122,9 @@ init_willmsg([{retain, Retain} | Opts], WillMsg) when is_boolean(Retain) ->
 init_willmsg([_Opt | Opts], State) ->
     init_willmsg(Opts, State).
 
+refresh_password([{password_refresher, PasswordRefresher}], State) when is_function(PasswordRefresher) ->
```
Comment (picaoao) [inline src/emqttc_protocol.erl]: what happens when PasswordRefresher os `undefined` ?

Comment (leandronsp) [reply src/emqttc_protocol.erl]: makes sense, added a scenario for that

**Comment by leandronsp** [comment]
up @emqplus

**Pull request closed without merging**